pkg/controller/cluster/cluster_controller.go

```diff
@@ -797,12 +799,11 @@ func (c *clusterController) updateClusterCondition(cluster *clusterv1alpha1.Clus
 		if cond.Type == condition.Type {
 			if cond.Status == condition.Status {
 				needToUpdate = false
-				continue
-			} else {
-				newConditions = append(newConditions, cond)
+				break
```

Comment:
If we break from here, conditions behind `cond` will be deleted. I think we should refactor this function like this. 

Copy all of the conditions to the `newConditions` except those type equal to the argument condition, then add argument condition.
```go
func (c *clusterController) updateClusterCondition(cluster *clusterv1alpha1.Cluster, condition clusterv1alpha1.ClusterCondition) {
	if cluster.Status.Conditions == nil {
		cluster.Status.Conditions = make([]clusterv1alpha1.ClusterCondition, 0)
	}

	newConditions := make([]clusterv1alpha1.ClusterCondition, 0)
	for _, cond := range cluster.Status.Conditions {
		if cond.Type == condition.Type {
			continue
		} 
		newConditions = append(newConditions, cond)
	}

	newConditions = append(newConditions, condition)
	cluster.Status.Conditions = newConditions
}
```

Reply:
We break here and set `needtoupdate` as `false`, so we will not update the conditions. It seems that your suggestion is more simple, which way should we choose?

Reply:
I prefer the simple one :)

Reply:
Squashed the commits.